src/auth/User.h

```diff
@@ -66,10 +66,15 @@ class User : public RefCountable
     const SBuf userKey() const {return userKey_;}
 
     /**
-     * How long these credentials are still valid for.
-     * Negative numbers means already expired.
+     * Credentials are not permitted to be re-used from a
```

Comment:
Can we say that credentials are not permitted to be cached "unless the scheme defines..."? In other words, caching code should not cache credentials for schemes that do not override this method. That would be a stronger rule/claim than the proposed "not permitted to be re-used".